src/charm.py

```diff
@@ -137,6 +148,28 @@ def __init__(self, *args: Any) -> None:
 
         self.endpoints_provider = KratosEndpointsProvider(self)
 
+        self.metrics_endpoint = MetricsEndpointProvider(
+            self,
+            relation_name=self._prometheus_scrape_relation_name,
+            jobs=[
+                {
+                    "metrics_path": "/metrics/prometheus",
+                    "static_configs": [
+                        {
+                            "targets": ["*:4434"],
+                        }
+                    ],
+                }
+            ],
+        )
+
+        self.loki_consumer = LogProxyConsumer(
+            self,
+            log_files=[self._log_path],
+            relation_name=self._loki_push_api_relation_name,
```

Comment:
**Nitpick:** You're using the default relation name in metadata ("logging", which is great!) so you don't _have_ to provided it here.

Reply:
It's consistent with the rest of the charm, and I prefer it because the docs in loki were a bit misleading/confusing: I specifically mean the loki_push_api lib line 46.